views/js/front/order-opc.js

```diff
@@ -0,0 +1,139 @@
+var isPudoPointSelected = false;
+
+$(document).ready(function (){
+
+    $(document).on('change', '.dpd-phone-block', function(){
+        handlePhoneNumber($(this));
+    });
+
+    $(document).on('click','.payment_module a', function (e){
+        e.preventDefault();
+
+        if ($('.dpd-phone-block') !== undefined) {
+            if(!handlePhoneNumber($('.dpd-phone-block'))) {
+                return;
+            }
+        }
+        if (!isPudoValid()) {
+            showError(order_opc_errors['pickup_point_error']);
+            return;
+        }
+        hideError();
+        location.href = $(this).attr('href');
+    });
+});
+
+function handlePhoneNumber(selector)
+{
+    if (!$('.dpd-phone-block')) {
+        return true;
+    }
+    var phone = selector.find('input[name="dpd-phone"]').val();
+    var phoneArea = selector.find('select[name="dpd-phone-area"] option:selected').val();
+
+    if (!validatePhoneNumber(phone, selector)) {
+        return false;
+    }
+
+    hideError();
+    selector.find('input[name="dpd-phone"]').css('border-color', 'initial');
+
+    saveSelectedPhoneNumber(phone, phoneArea)
+
+    return true;
+}
+
+function saveSelectedPhoneNumber(phoneNumber, phoneArea) {
+    $.ajax(dpdHookAjaxUrl, {
+        type: 'POST',
+        data: {
+            'ajax': 1,
+            'phone_number': phoneNumber,
+            'phone_area': phoneArea,
+            'action': 'saveSelectedPhoneNumber',
+            'token': static_token
+        },
+        success: function (response) {
+        },
+        error: function (response) {
+
+        }
```

Comment:
what happens if I send invalid phone number ?

Reply:
I suppose you should handle both success and error cases ?

Reply:
I show message in network now does not reach front as I already did front validation not to execute script if invalid Atm when number is invalid i stops click event and backend throws invalid phone number never saves phone